### PR TITLE
Use py-build-cmake

### PR DESCRIPTION
**Pull request by ilya-lavrenov**

### CODE REVIEW

File: src/python/CMakeLists.txt [hunk 1]
```diff
@@ -40,12 +41,28 @@ if(rpaths)
 endif()
 
 find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
-install(FILES "${CMAKE_BINARY_DIR}/openvino_genai/__init__.py" "${CMAKE_BINARY_DIR}/openvino_genai/__version__.py" DESTINATION python/openvino_genai/ COMPONENT pygenai_${Python_VERSION_MAJOR}_${Python_VERSION_MINOR})
-install(TARGETS openvino_genai py_generate_pipeline LIBRARY DESTINATION python/openvino_genai/ COMPONENT pygenai_${Python_VERSION_MAJOR}_${Python_VERSION_MINOR})
+install(FILES "${OpenVINOGenAI_BINARY_DIR}/openvino_genai/__init__.py"
```
Comment (Wovchena) [inline src/python/CMakeLists.txt]: `wheel.packages = ["src/python/openvino_genai"]` used to install \_\_init__.py into wheel. But how does it work now? :) init is installed although there's nothing doing that

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: src/python/CMakeLists.txt [hunk 2]
```diff
@@ -15,12 +15,13 @@ if(NOT pybind11_POPULATED)
 endif()
 
 pybind11_add_module(py_generate_pipeline py_generate_pipeline.cpp)
-target_link_libraries(py_generate_pipeline PRIVATE openvino::genai nlohmann_json::nlohmann_json)
+target_link_libraries(py_generate_pipeline PRIVATE openvino::genai)
 set_target_properties(py_generate_pipeline PROPERTIES
-    LIBRARY_OUTPUT_DIRECTORY "$<1:${CMAKE_BINARY_DIR}/openvino_genai/>"
+    LIBRARY_OUTPUT_DIRECTORY "$<1:${OpenVINOGenAI_BINARY_DIR}/openvino_genai/>"
 )
-file(COPY "${CMAKE_CURRENT_SOURCE_DIR}/openvino_genai/__init__.py" DESTINATION "${CMAKE_BINARY_DIR}/openvino_genai/")
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/python/CMakeLists.txt:22]: reverted

File: src/python/CMakeLists.txt [hunk 3]
```diff
@@ -15,12 +15,13 @@ if(NOT pybind11_POPULATED)
 endif()
 
 pybind11_add_module(py_generate_pipeline py_generate_pipeline.cpp)
-target_link_libraries(py_generate_pipeline PRIVATE openvino::genai nlohmann_json::nlohmann_json)
+target_link_libraries(py_generate_pipeline PRIVATE openvino::genai)
 set_target_properties(py_generate_pipeline PROPERTIES
-    LIBRARY_OUTPUT_DIRECTORY "$<1:${CMAKE_BINARY_DIR}/openvino_genai/>"
+    LIBRARY_OUTPUT_DIRECTORY "$<1:${OpenVINOGenAI_BINARY_DIR}/openvino_genai/>"
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/python/CMakeLists.txt]: reverted

File: src/python/CMakeLists.txt (hunk 1 above)
Comment (ilya-lavrenov) [reply src/python/CMakeLists.txt]:
```
[tool.py-build-cmake.module]
directory = "src/python"
```

File: .github/workflows/genai_python_lib.yml [hunk 4]
```diff
@@ -78,8 +74,8 @@ jobs:
...
       - run: call ./ov/setupvars.bat && cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
       - run: call ./ov/setupvars.bat && cmake --build ./build/ --config Release -j
-      - run: call ./ov/setupvars.bat && python -m pip install ./thirdparty/openvino_tokenizers/[transformers] -r ./requirements-build.txt -r ./tests/python_tests/requirements.txt --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/pre-release --upgrade-strategy eager
+      - run: call ./ov/setupvars.bat && python -m pip install ./thirdparty/openvino_tokenizers/[transformers] -r ./tests/python_tests/requirements.txt --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/pre-release --upgrade-strategy eager
       # cmd evaluates variables in a different way. Setting PYTHONPATH before setupvars.bat instead of doing that after solves that.
       - run: set "PYTHONPATH=./build/" && call ./ov/setupvars.bat && python -m pytest ./tests/python_tests/test_generate_api.py -m precommit
-      - run: call ./ov/setupvars.bat && python -m pip install . --config-settings=build-dir="build" --verbose
+      - run: call ./ov/setupvars.bat && python -m pip install . --verbose
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply .github/workflows/genai_python_lib.yml:80]: this variable is not used by new approach as it builds to its internal directory.

**Review by Wovchena — APPROVED** [review]

**Pull request merged**